`_linalg_svd`: change to new `MetaBase` API.

ysiraichi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #78039
* #78038
* __->__ #77540
* #77539
* #76098
* #76097

This PR modifies `_linalg_svd` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Test for svd

ysiraichi committed:
`_linalg_svd`: change to new `MetaBase` API.

This PR modifies `_linalg_svd` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Test for svd

[ghstack-poisoned]

Lezcano commented on aten/src/ATen/native/BatchLinearAlgebra.cpp:
```diff
@@ -3493,33 +3493,11 @@ TORCH_IMPL_FUNC(_linalg_svd_out)(const Tensor& A,
-  // Prepare U / Vh
-  // U_ and Vh_ are just going to be accessed whenever compute_uv == true
-  const auto U_ready = !compute_uv || U.mT().is_contiguous();
-  const auto U_ = borrow_else_clone(U_ready, U, U, /*C-contig*/false);
```
Nit. If you go on and move `ldl_factor` to use this, we would have all the operations that correctly handled strides using this pattern and we could remove `borrow_else_clone`. Then, I could port the new ones (https://github.com/pytorch/pytorch/pull/77634) to use this pattern.

lezcano reviewed (approved):
As commented in the other, I believe the test is not necessary. Otherwise this LGTM

ysiraichi committed:
Rebase. on "`_linalg_svd`: change to new `MetaBase` API."


This PR modifies `_linalg_svd` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Test for svd

[ghstack-poisoned]

ysiraichi committed:
Remove redundant test. on "`_linalg_svd`: change to new `MetaBase` API."


This PR modifies `_linalg_svd` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Test for svd

[ghstack-poisoned]

ysiraichi committed:
Fix new 'set_output' that wasn't replaced. on "`_linalg_svd`: change to new `MetaBase` API."


This PR modifies `_linalg_svd` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Test for svd

[ghstack-poisoned]

ysiraichi committed:
Remove unused 'use_cusolver' variable. on "`_linalg_svd`: change to new `MetaBase` API."


This PR modifies `_linalg_svd` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Test for svd

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge this
